.Lib9c.Tests/Action/Scenario/ItemCraftTest.cs

```diff
@@ -0,0 +1,405 @@
+            var maxUnlockStage = recipeList.Aggregate(0, (e, c) => Math.Max(e, c.UnlockStage));
+            var unlockRecipeIdsAddress = _avatarAddr.Derive("recipe_ids");
+            var recipeIds = List.Empty;
+            for (int i = 1; i < maxUnlockStage + 1; i++)
+            {
+                recipeIds = recipeIds.Add(i.Serialize());
+            }
+
+            stateV2 = stateV2.SetState(unlockRecipeIdsAddress, recipeIds);
+
+            // Prepare combination slot
+            for (var i = 0; i < targetItemIdList.Length; i++)
+            {
+                stateV2 = CraftUtil.PrepareCombinationSlot(stateV2, _avatarAddr, i);
+            }
+
+            // Initial inventory must be empty
+            var inventoryState = new Inventory((List)stateV2.GetState(_inventoryAddr));
+            Assert.Equal(0, inventoryState.Items.Count);
+
+            // Add materials to inventory
+            stateV2 = CraftUtil.AddMaterialsToInventory(
+                stateV2,
+                _tableSheets,
+                _avatarAddr,
+                allMaterialList,
+                random
+            );
+
+            for (var i = 0; i < recipeList.Count; i++)
+            {
+                // Unlock stage
+                var equipmentRecipe = recipeList[i];
+                stateV2 = CraftUtil.UnlockStage(
+                    stateV2,
+                    _tableSheets,
+                    _worldInformationAddr,
+                    equipmentRecipe.UnlockStage
+                );
+
+                // Do Combination Action
+                var action = new CombinationEquipment
+                {
+                    avatarAddress = _avatarAddr,
+                    slotIndex = i,
+                    recipeId = equipmentRecipe.Id,
+                    subRecipeId = equipmentRecipe.SubRecipeIds?[0],
+                };
+
+                stateV2 = action.Execute(new ActionContext
+                {
+                    PreviousStates = stateV2,
+                    Signer = _agentAddr,
+                    BlockIndex = 0L,
+                    Random = random,
+                });
+                var slotState = stateV2.GetCombinationSlotState(_avatarAddr, i);
+                // TEST: requiredBlock
+                // TODO: Check reduced required block when pet comes in
+                Assert.Equal(equipmentRecipe.RequiredBlockIndex, slotState.RequiredBlockIndex);
+            }
+
+            inventoryState = new Inventory((List)stateV2.GetState(_inventoryAddr));
+            // TEST: Only created equipments should remain in inventory
+            Assert.Equal(recipeList.Count, inventoryState.Items.Count);
+            foreach (var itemId in targetItemIdList)
+            {
+                // TEST: Created equipment should match with targetItemList
+                Assert.NotNull(inventoryState.Items.Where(e => e.item.Id == itemId));
+            }
+        }
+
+        [Theory]
+        [InlineData(1, new[] { 200000 })] // 맛없는 요리
```

Comment:
I think the test result of this inline data row is not helpful. Can you consider adding craftable recipe?

Reply:
> I think the test result of this inline data row is not helpful. Can you consider adding craftable recipe?

Oh, I just rechecked `ConsumableItemRecipeSheet` and 20`1`000 was what I originally attended.
Thanks for pointing out!